feeds/pypi/pypi.go

```diff
@@ -72,20 +86,94 @@ func fetchPackages() ([]*Package, error) {
 	return rssResponse.Packages, nil
 }
 
-type Feed struct{}
+func fetchCriticalPackages(packageList []string) ([]*Package, error) {
+	responseChannel := make(chan *Response)
+	errs := make(chan error)
+	var wg sync.WaitGroup
+
+	for _, pkgName := range packageList {
+		wg.Add(1)
+		go func(pkgName string) {
+			resp, err := httpClient.Get(fmt.Sprintf(packageURLFormat, pkgName))
+			if err != nil {
+				errs <- err
+				return
+			}
+			defer resp.Body.Close()
+			rssResponse := &Response{}
+			err = xml.NewDecoder(resp.Body).Decode(rssResponse)
+			if err != nil {
+				errs <- err
+				return
+			}
+			wg.Done()
+			responseChannel <- rssResponse
+		}(pkgName)
+	}
+	wg.Wait()
+	select {
+	case err := <-errs:
+		return nil, err
+	default:
+		close(errs)
+	}
+
+	pkgs := []*Package{}
+	for i := 0; i < len(packageList); i++ {
+		select {
+		case response := <-responseChannel:
+			pkgs = append(pkgs, response.Packages...)
+		default:
+			return nil, errMissingParallelPackageResponse
+		}
+	}
+	return pkgs, nil
+}
+
+type Feed struct {
+	packages *[]string
+}
+
+func New(feedOptions feeds.FeedOptions) (*Feed, error) {
+	return &Feed{
+		packages: feedOptions.Packages,
+	}, nil
+}
 
 func (feed Feed) Latest(cutoff time.Time) ([]*feeds.Package, error) {
 	pkgs := []*feeds.Package{}
-	pypiPackages, err := fetchPackages()
+	var pypiPackages []*Package
+	var err error
+
+	if feed.packages == nil {
```

Comment:
For a scenario where somebody is interested in the firehose as-well as set of specific packages, the expectation would be with this implementation that they'd have to run two instances of the binary? This make the configuration logic clean, but do we want to consider allowing either/or here for a single instance?

Reply:
Currently it is only possible to have one feed of each type. In a previous PR this was possible but increased complexity of the code and configuration. We should see if there is much interest in this before looking to implement this.

Reply:
Sure, makes sense to tackle it if requested